app/controllers/uploads_controller.rb

```diff
@@ -9,15 +9,32 @@ class UploadsController < ApplicationController
   protect_from_forgery except: :show
 
   before_action :is_asset_path, :apply_cdn_headers, only: [:show, :show_short, :show_secure]
-  before_action :external_store_check, only: [:show_secure, :generate_presigned_put, :complete_external_upload]
+  before_action :external_store_check, only: [
+    :show_secure,
+    :generate_presigned_put,
+    :complete_external_upload,
+    :create_multipart,
+    :batch_presign_multipart_parts,
+    :abort_multipart,
+    :complete_multipart
+  ]
+  before_action :can_upload_external?, only: [:create_multipart, :generate_presigned_put]
 
   SECURE_REDIRECT_GRACE_SECONDS = 5
-  PRESIGNED_PUT_RATE_LIMIT_PER_MINUTE = 5
+  PRESIGNED_PUT_RATE_LIMIT_PER_MINUTE = 10
```

Comment:
I suspect we will need to tune these rate limits, I thought that these were just a good starting point.